sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1643,19 +1680,34 @@ def _provision(self,
                 tpu_create_script=config_dict.get('tpu-create-script'),
                 tpu_delete_script=config_dict.get('tpu-delete-script'))
 
-            # Get actual zone info and save it into handle
-            get_zone_cmd = handle.launched_resources.cloud.get_zone_shell_cmd()
-            if get_zone_cmd is not None:
-                # We leave the zone field to None for multi-node cases
-                # if zone is not specified because head and worker nodes
-                # can be launched in different zones.
-                if (task.num_nodes == 1 or
-                        handle.launched_resources.zone is not None):
-                    returncode, stdout, _ = self.run_on_head(
-                        handle, get_zone_cmd, require_outputs=True)
-                    # zone will be checked during Resources cls initialization.
-                    handle.launched_resources = handle.launched_resources.copy(
-                        zone=stdout.strip())
+            # Get actual zone info and save it into handle.
+            if (handle.launched_nodes == 1 and
+                    handle.launched_resources.cloud.is_same_cloud(
```

Comment:
Instead of checking this statement, do you think we can directly check `if len(config_dict['zones']):`?

Reply:
Good catch! Simplified the logic now.